curriculum/challenges/english/25-front-end-development/review-react-basics/67487e141bb6a7140a352e12.md

```diff
@@ -0,0 +1,253 @@
+npx create-react-app my-react-app
+```
+
+- For Vue.js projects, the Vue CLI is a popular choice. You can start a new Vue project by running:
+
+```bash
+npm init vue@latest
+```
+
+This command will prompt you with questions about your project setup, allowing you to choose features like TypeScript support, routing, and state management.
+
+- Angular projects can be set up using the Angular CLI. To create a new Angular project, you can run:
+
+```bash
+ng new my-angular-app
+```
+
+- The Angular CLI will ask you some questions to customize your project setup, such as if you want to include routing and which CSS preprocessor you would like to use.
+
+## Dependency management in React
+
+- Dependency management refers to the process of handling external code packages that your project relies on. These packages, also known as dependencies, can range from small utility functions to large libraries like React JS.
+- Dependency management is typically handled by package managers. The most popular package managers are npm (Node Package Manager), Yarn, and more recently, pnpm.
+- When you create a new React project using Create React App, npm is used to install the necessary dependencies. You can add new dependencies to your project using `npm install <package-name>`. This will download the package and add it to your project's `package.json` file.
+- The `package.json` file contains a list of all the dependencies and devDependencies used in your project, along with their versions. This file is used to track and manage the dependencies of your project.
+
+Here is an example of a package.json file:
+
+```json
+{
+    "name": "my-react-app",
+    "version": "1.0.0",
+    "dependencies": {
+        "react": "^17.0.2",
+        "react-dom": "^17.0.2"
+    },
+    "devDependencies": {
+        "babel": "^7.15.0",
+        "webpack": "^5.51.1"
+    }
+}
+```
+
+Dependency management also involves keeping your packages up-to-date. You can check for updates using commands like `npm outdated` and update packages using `npm update`. It's important to be cautious when updating dependencies, especially for major version changes, as these can most of the time break your codebase.
+
+Here are some npm commands that are commonly used for dependency management:
+
+- `npm install`: installs all the dependencies listed in the package.json file.
+- `npm install <package-name>`: installs a specific package and adds it to the package.json file.
+- `npm update`: updates all the dependencies to their latest versions.
+
+## Passing props in React components
+
+- In React, props (short for properties) are a way to pass data from a parent component to a child component. This mechanism is needed to create reusable and dynamic UI elements.
+- Props can be of these types: strings, numbers, booleans, objects, or arrays. To pass props from a parent to a child component, you add the props as attributes when you use the child component in the parent's JSX. Here's a simple example:
```

Comment:
```suggestion
- Props can be any JavaScript value. To pass props from a parent to a child component, you add the props as attributes when you use the child component in the parent's JSX. Here's a simple example:
```
Maybe include examples, too, but just including examples implies that they're the only allowed values.